### PR TITLE
[Tanks] Eye of F'harg / Eye of Shatug (Trinket)

### PR DESCRIPTION
**Pull request by joshinat0r**
Adds the average stats provided from the trinket to the item-breakdown. Initial trinket state is in the bonusIDs of the trinket (3618 for the stamina one, 3619 for the armor one) and switches upon cast.

[Log](http://localhost:3000/report/72Ddcrqay1gKP4pN/57-LFR+Argus+the+Unmaker+-+Kill+(12:02)/282-Madnéss) with several switches, pull with stamina-version
[Log](http://localhost:3000/report/DqrG32f1Z6nxAb4v/1-LFR+Aggramar+-+Kill+(5:06)/11-Alaeya) with several switches, pull with agility-version
[Log](http://localhost:3000/report/CjZw9tc4ByTDVNvF/25-Mythic+Argus+the+Unmaker+-+Kill+(10:31)/35-Myrei) with no switches, pull with agility-version
[Log](http://localhost:3000/report/MjHgGCBWfd21Zynv/13-Mythic+Argus+the+Unmaker+-+Wipe+13+(1:34)/9-Baelath) with no switches, pull with stamina-version

eye2
eye1
eye4
eye3

### CODE REVIEW

File: src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/EyeOfHounds.js [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+import { calculatePrimaryStat, calculateSecondaryStatDefault } from 'common/stats';
+import Abilities from 'Parser/Core/Modules/Abilities';
+
+/**
+ * Eye of Shatug / Eye of F'harg
+ * Use: swaps between both versions
+ * initial state is in the trinkets bonusID (3618 for the stamina one, 3619 for the armor one)
+*/
+
+const ARMOR_EYE = 3619;
```
Comment (MartijnHols) [inline src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/EyeOfHounds.js]: This var name is kinda unclear; I thought it was about armor rating.

File: src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/EyeOfHounds.js [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+class DiimasGlacialAegis extends Analyzer {
+  static dependencies = {
+    combatants: Combatants,
+    abilities: Abilities,
+  };
+
+  isCurrentlyVers = true;
+  lastSwap = 0;
+  
+  stam_stat = 0;
```
Comment (MartijnHols) [inline src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/EyeOfHounds.js]: Please use camelCasing style for properties

File: src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/EyeOfHounds.js [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      this.str_uptime += (event.timestamp - this.lastSwap);
+    }
+
+    this.lastSwap = event.timestamp;
+    this.isCurrentlyVers = !this.isCurrentlyVers;
+  }
+
+  on_finished(event) {
+    if (this.isCurrentlyVers) {
+      this.stam_uptime += (this.owner.fight.end_time - this.lastSwap);
```
Comment (MartijnHols) [inline src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/EyeOfHounds.js]: This will lead to inaccuracies during partial parses. Please make a public getter for getting the uptime that does this whenever the data is read.

File: src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/EyeOfHounds.js [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+  strUptime = 0;
+
+  on_initialized() {
+    this.active = this.combatants.selected.hasTrinket(ITEMS.EYE_OF_HOUNDS.id);
+    if (this.active) {
+      const iLvl = this.combatants.selected.getItem(ITEMS.EYE_OF_HOUNDS.id).itemLevel;
+      const bonusIDs = Object.values(this.combatants.selected.gear).filter(item => item.id === ITEMS.EYE_OF_HOUNDS.id)[0].bonusIDs;
+      
+      if (bonusIDs.includes(ARMOR_EYE_BONUS_ID)) {
+        this.isCurrentlyVers = false;
```
Comment (Gebuz) [inline src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/EyeOfHounds.js]: You could shorten this to `this.isCurrentlyVers = !bonusIDs.includes(ARMOR_EYE_BONUS_ID);`. And if you want it a bit prettier without the invert, you could check for the vers bonus ID instead.

File: src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/EyeOfHounds.js [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+      this.strUptime += event.timestamp - this.lastCheck;
+    }
+
+    this.lastCheck = event.timestamp;
+    this.isCurrentlyVers = !this.isCurrentlyVers;
+  }
+
+  get staminaUptime() {
+    //add uptime if stamina-version is active
+    if (this.isCurrentlyVers === true) {
```
Comment (Gebuz) [inline src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/EyeOfHounds.js]: `=== true` is redundant. Same with `=== false` in strengthUptime, invert the value instead.

File: src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/EyeOfHounds.js [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+import { calculatePrimaryStat, calculateSecondaryStatDefault } from 'common/stats';
+import Abilities from 'Parser/Core/Modules/Abilities';
+
+/**
+ * Eye of Shatug / Eye of F'harg
+ * Use: swaps between both versions
+ * initial state is in the trinkets bonusID (3618 for the stamina one, 3619 for the armor one)
+*/
+
+const ARMOR_EYE_BONUS_ID = 3619;
```
Comment (Gebuz) [inline src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/EyeOfHounds.js]: It's a bit confusing that you call this armor  and `isCurrentlyVers` vers when everything else uses stam/str in the names. Consider renaming one or the other so everything is vers/armor or stam/str.

File: src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/EyeOfHounds.js [hunk 7]
```diff
@@ -0,0 +1,139 @@
...
+
+    return this.strUptime / (this.owner.currentTimestamp - this.owner.fight.start_time);
+  }
+
+  get averageStats() {
+    const itemBreakdown = [];
+    const stam = this.staminaUptime;
+    const str = this.strengthUptime;
+
+    if (stam !== 0) {
```
Comment (Gebuz) [inline src/Parser/Core/Modules/Items/Legion/AntorusTheBurningThrone/EyeOfHounds.js]: `!== 0` is redundant since 0 evaluates to false. Same for str below.
